packages/video_player_avplayer/tizen/src/plus_player.cc

```diff
@@ -0,0 +1,608 @@
+  if (player_->GetState() < plusplayer::State::kReady) {
+    LOG_ERROR("[PlusPlayer] Player is not ready.");
+    return;
+  }
+
+  if (on_seek_completed_) {
+    LOG_ERROR("[PlusPlayer] Player is already seeking.");
+    return;
+  }
+
+  on_seek_completed_ = std::move(callback);
+  if (!player_->Seek(position)) {
+    on_seek_completed_ = nullptr;
+    LOG_ERROR("[PlusPlayer] Fail to seek.");
+  }
+}
+
+int32_t PlusPlayer::GetPosition() {
+  uint64_t position = 0;
+  plusplayer::State state = player_->GetState();
+  if (state == plusplayer::State::kPlaying ||
+      state == plusplayer::State::kPaused) {
+    if (!player_->GetPlayingTime(&position)) {
+      LOG_ERROR("[PlusPlayer] Fail to get the current playing time.");
+    }
+  }
+  return position;
+}
+
+int32_t PlusPlayer::GetDuration() {
+  int64_t duration = 0;
+  if (player_->GetState() >= plusplayer::State::kTrackSourceReady) {
+    if (!player_->GetDuration(&duration)) {
+      LOG_ERROR("[PlusPlayer] Fail to get the duration.");
+    }
+    LOG_INFO("[PlusPlayer] Video duration: %llu.", duration);
+  }
+  return duration;
+}
+
+void PlusPlayer::GetVideoSize(int32_t *width, int32_t *height) {
+  if (player_->GetState() >= plusplayer::State::kTrackSourceReady) {
+    bool found = false;
+    std::vector<plusplayer::Track> tracks = player_->GetActiveTrackInfo();
+    for (auto track : tracks) {
+      if (track.type == plusplayer::TrackType::kTrackTypeVideo) {
+        *width = track.width;
+        *height = track.height;
+        found = true;
+      }
+    }
+    if (!found) {
+      LOG_ERROR("[PlusPlayer] Failed to get video size.");
+    } else {
+      LOG_INFO("[PlusPlayer] Video widht: %d, height: %d.", *width, *height);
+    }
+  }
+}
+
+bool PlusPlayer::isReady() {
+  return plusplayer::State::kReady == player_->GetState();
+}
+
+bool PlusPlayer::SetDisplay() {
+  int x = 0, y = 0, width = 0, height = 0;
+  ecore_wl2_window_proxy_->ecore_wl2_window_geometry_get(native_window_, &x, &y,
+                                                         &width, &height);
+  int surface_id =
+      ecore_wl2_window_proxy_->ecore_wl2_window_surface_id_get(native_window_);
+  if (surface_id < 0) {
+    LOG_ERROR("[PlusPlayer] Failed to get surface id.");
+    return false;
+  }
+  bool ret = player_->SetDisplay(plusplayer::DisplayType::kOverlay, surface_id,
+                                 x, y, width, height);
+  if (!ret) {
+    LOG_ERROR("[PlusPlayer] Failed to set display.");
+    return false;
+  }
+
+  ret = player_->SetDisplayMode(plusplayer::DisplayMode::kDstRoi);
+  if (!ret) {
+    LOG_ERROR("[PlusPlayer] Failed to set display mode.");
+    return false;
+  }
+
+  return true;
+}
+
+flutter::EncodableList PlusPlayer::getTrackInfo(int32_t track_type) {
+  if (!player_) {
+    LOG_ERROR("[PlusPlayer] Player not created.");
+    return {};
+  }
+
+  plusplayer::State state = player_->GetState();
+  if (state == plusplayer::State::kNone || state == plusplayer::State::kIdle) {
```

Comment:
According to the document, it is not allowed when the state is kTypeFinderReady, but I haven't found a case where the state is kTypeFinderReady. (I don't know what type finder is.) If you don't have the kTypeFinderReady case, please ignore this comment.
```
   * @pre       The player can be one of #State::kTrackSourceReady,
   * #State::kReady, #State::kPlaying or #State::kPaused
```

Reply:
Fixed.
[Codes cleanup](https://github.com/xiaowei-guan/plugins/pull/6/commits/5f09b1831625607e9677d98d35788ac9db34b2c7)